base/sort.jl

```diff
@@ -426,6 +426,23 @@ struct InsertionSortAlg <: Algorithm end
 struct QuickSortAlg     <: Algorithm end
 struct MergeSortAlg     <: Algorithm end
```

Comment:
Would it make sense/be possible to add a `RadixSortAlg` type? Maybe it's not super useful as it's used by default, but it would be good for consistency and e.g. for benchmarking (to ensure the intended algorithm is used). Of course it would have to throw an error for types that are not supported.

Reply:
There are simple common reasons for folks to manually dispatch to `QuickSort` instead of `MergeSort` (if they value performance over stability), but I see no such use case for manual choice of radix sort. 

I was following the patter of counting sort (`sort_int_range!`) which, like radix sort, is dynamically dispatched to, requires some precomputed values, and cannot be user specified by the `alg` keyword.

Right now, the necessary preprocessing is interleaved with the decisions about which algorithm to dispatch to. It's possible to add a `RadixSortAlg` which forces those dispatch decisions but this is much harder than it is for `QuickSort` and `MergeSort`.

If we ever want to remove radix sort from base (e.g. a better non-comparison based algorithm comes up), it will be nice to never have defined a user-facing `RadixSorAlgt` type. 

`RadixSortAlg` would be convenient to have for benchmarking, as would `CountSortAlg`, but I think those methods belong in `contrib/` or `SortPerf.jl`, not `base/sort.jl`.

Reply:
Tentatively, I think we should move the radix sort chunk size heuristic and switch based dispatch out of `sort!(... ::AdaptiveSort ...)` so that other folks can use it in their benchmarking definitions of `sort!(... ::RadixSortAlg ...)`, but not define `RadixSortAlg` in base.

Reply:
Done. The `chucnk_size` argument to `radix_sort!` is now optional.

Reply:
OK. Let's hear what others think.

Reply:
@oscardssmith, what do you think about adding a `RadixSortAlg` type?

Reply:
What did we decide here? Close this review?

Reply:
I think it might be useful to have `NonCompareSort` or something similar that doesn't tie us to using `RadixSort` forever, but I also don't really care as long as the thing we give to users by default is fast.